dandiapi/api/models/asset.py

```diff
@@ -203,35 +203,32 @@ def is_zarr(self):
     def size(self):
         if self.is_blob:
             return self.blob.size
-        elif self.is_embargoed_blob:
+        if self.is_embargoed_blob:
             return self.embargoed_blob.size
-        else:
-            return self.zarr.size
+        return self.zarr.size
```

Comment:
I think this is a good example of this rule being kind of bad 🤷‍♂️.

Reply:
I think overall the design pattern above is suboptimal regardless of the diff which is overall "potato vs potato" to me. Neither original nor new code is provisioning to have some case which is no longer fulfilling the design (e.g. when new types of assets would be added...).  Should have been likely a sequence of `if/elif/else: raise AssertionError(f"Do not know how to get size for the {self}")`.